microprofile-health/README.adoc

```diff
@@ -0,0 +1,424 @@
+include::../shared-doc/attributes.adoc[]
+
+= microprofile-health: MicroProfile Health QuickStart
+:author: Martin Stefanko
+:level: Beginner
+:technologies: MicroProfile Health
+
+[abstract]
+The `microprofile-health` quickstart demonstrates the use of the MicroProfile Health
+specification in {productName}.
+
+:standalone-server-type: default
+:archiveType: jar
+:archiveName: {artifactId}-service
+
+== What is it?
+
+MicroProfile Health allows applications to provide information about their state
+to external viewers which is typically useful in cloud environments where automated
+processes must be able to determine whether the application should be discarded
+or restarted.
+
+== Architecture
+
+In this quickstart, we have a simple REST application that exposes MicroProfile Health
+functionalities at the `/health/live` and `/health/ready` endpoints according to the
+specification.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the
+application step by step. However, you can go right to the completed example which
+is available in this directory.
+
+. Start your {productName} server
+
+. Compile the application code and deploy it to {productName}
+[source,options="nowrap"]
+----
+$ mvn clean package wildfly:deploy
+----
+
+// Start the {productName} Standalone Server
+include::../shared-doc/start-the-standalone-server.adoc[leveloffset=+1]
+
+=== Running the health check
+
+The {productName} server directly exposes three REST endpoints:
+
+- `/health/live` - The application is up and running.
+- `/health/ready` - The application is ready to serve requests.
+- `/health` - Accumulating all health check procedures in the application.
+
+To check that the {productName} is working as expected:
+
+* access the `http://localhost:9990/health/live` endpoint using your browser or
+`curl http://localhost:9990/health/live`
+
+All of the health REST endpoints return a simple JSON object with two fields:
+
+* `status` -- the overall result of all the health check procedures
+* `checks` -- an array of individual checks
+
+The general `status` of the health check is computed as a logical AND of all the
+declared health check procedures. The `checks` array is empty as we have not specified
+any health check procedure yet so let's define some.
+
+NOTE: Similarly you can access `http://localhost:9990/health/ready` and
+`http://localhost:9990/health`
+
+Congratulations! You successfully deployed and run this quickstart. If you wish to
+create the application step by step you can follow the subsequent sections.
+
+== Creating the Maven Project
+
+[source,options="nowrap"]
+----
+mvn archetype:generate \
+    -DgroupId=org.wildfly.quickstart.microprofile.health \
```

Comment:
Existing quickstarts use a groupID of "org.wildfly.quickstarts" should the projects we generate also use this groupID?

Reply:
Actually, it seems they use `org.jboss.as.quickstarts`. @emmartins should we change packages to `org.jboss.as.quickstarts`?

Reply:
For now, updated to `org.wilfdly.quickstarts`.

Reply:
please use same maven group id as others